WalletWasabi/Tor/TorMonitor.cs

```diff
@@ -39,7 +39,15 @@ public TorMonitor(TimeSpan period, Uri fallbackBackendUri, TorHttpClient httpCli
 		private TorHttpClient HttpClient { get; }
 		private TorProcessManager TorProcessManager { get; }
 
-		public async Task StartBootstrapMonitorAsync(CancellationToken appShutdownToken)
+		/// <inheritdoc/>
+		public override Task StartAsync(CancellationToken cancellationToken)
+		{
+			_ = StartBootstrapMonitorAsync(cancellationToken);
```

Comment:
Looks much better now. Is there a reason why it's not awaited? I think you can make the `StartAsync` method async and await this function.

In that case:

- `HostedServices.StartAllAsync` would start it in parallel with other services.
- `HostedServices.StartAllAsync` would await it
- only very few services are present after `HostedServices.StartAllAsync`: `Synchronizer`, `TransactionBroadcaster`, `CoinJoinProcessor`, `RpcServer`, `blockProvider` and some later initialization parts of the wallet manager. I think those can well wait until Tor is established.
- finally stopping would then also wait for either successful cancellation of bootstrap monitoring or the end of the startasync, which is good for clean shutdown.

But I'm kinda doing 4D chess here, so if you considered awaiting or tried it and it didn't work well, then I guess I'm ok with it.

Reply:
> Is there a reason why it's not awaited?

I do not await it because `StartBootstrapMonitorAsync` is supposed to run for the lifetime of the WW application. So if I await it, it will never really return.

Reply:
Oh I see. Makes sense. Then you can create a new periodicrunner class and call its start function here. What do you think? If you can't, I'd like to give it a try myself and see why that wouldn't work.

Reply:
> Then you can create a new periodicrunner class and call its start function here.

`StartBootstrapMonitorAsync` method cannot be modeled as a periodic runner but we can certainly have `StartBootstrapMonitorAsync` in a separate class (which can by itself be a hosted service if you want). If I get it right, I can change it. It should be fairly easy.

Reply:
Looks to me `StartBootstrapMonitorAsync` maps to periodcrunner where the period is `TimeSpan.Zero` and the `ExecuteAsync(CancellationToken stoppingToken)`'s `stoppingToken` maps to `linkedCts`?

Again, I may be wrong and only figure it out if I try. Do you see what I'm missing if I'm missing something?

Reply:
Well, I would use `PeriodicRunner` for something which is periodic by nature. `StartBootstrapMonitorAsync` uses a simple while cycle because you don't know when Tor will crash or if it will crash at all ([line](https://github.com/zkSNACKs/WalletWasabi/pull/6180/files#diff-ccd1447a84db8253a1d8dd6136f6426c520eacb9abb699ea7296e375ea133fd7R58)). So I would rather not use `PeriodicRunner` here because conceptually I think we model something different here (and yes, it may be in a "mathematic" way similar but then for me it's harder to understand).

If your goal is to generalize the code so that we can write a new Tor monitor (like a watcher of Tor circuits or whatever.) more easily, we can then create a new class that wraps the basic "monitor restarting" in a separate class.

Also this PR received quite a lot of testing so far and it seems to work correctly. I would not like to lose that. Meaning, I would rather modify it in a next PR, if required.

Reply:
I suppose my main concern is the fire and forgetting of a nearly forever running async task. These sometimes lead to shutdown hanging and impossible to debug. Is it possible at least to await this task somewhere?

Reply:
I see. Makes sense. I have added a task that is awaited now plus a logging command so that we know if the bootstrap monitor gets stuck.